### PR TITLE
[PS4/PS5][Driver][DWARF] Always emit .debug_aranges for SCE tuning

### PR DESCRIPTION
**Pull request by playstation-edd**
Some of SIE's post-mortem analysis infrastructure currently makes use of .debug_aranges, so we'd like to ensure the section's presence in PlayStation binaries. The simplest way to do this is to force emission when the debugger tuning is set to SCE (which is in turn typically initialized from the target triple). This also simplifies the driver.

llvm/test/DebugInfo/debuglineinfo-path.ll has been marked as UNSUPPORTED on PlayStation. When aranges are emitted, the DWARF in the test case is such that relocations need to be applied to the aranges section in order for symbolization to work. An alternative approach would be to implement the application of relocations in DWARFDebugArangeSet. While experiments show that this can be made to work with a modest patch, the test cases would be rather contrived. Since I expect the only utility for such a change would be to make this test case pass for PlayStation targets, and few - if any - outside of PlayStation care about aranges, UNSUPPORTED would seem to be a more practical option.

This was originally commited as 22eb290a96 (#99629) and later reverted at 84658fb82b (#99711) due to test failures on SIE built bots. These failures shouldn't recur due to 3b24e5d450 (#99897) and the aforementioned change to debuglineinfo-path.ll.

SIE tracker: TOOLCHAIN-16951

### CODE REVIEW

File: llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+; CHECK:      .debug_aranges contents:
+; CHECK-NEXT: Address Range Header:
+; CHECK-SAME:   length = 0x0000002c,
+
+; IR generated and reduced from:
+; $ cat foo.c
+; int foo;
+; $ clang -g -S -emit-llvm foo.c -o foo.ll
+
+target triple = "x86_64-unknown-linux-gnu"
```
Comment (pogo59) [inline llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll:16]: This is redundant with the explicit triple on the llc command line.

**Review by pogo59 — APPROVED** [review]
One nit on the test, otherwise LGTM.

File: llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll [hunk 2]
```diff
@@ -0,0 +1,33 @@
+; This checks that .debug_aranges is always generated for the SCE debugger
```
Comment (MaskRay) [inline llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll:1]: The https://llvm.org/docs/TestingGuide.html#elaborated-tests style could make test update easier in the future.

File: llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll (hunk 1 above)
Comment (MaskRay) [reply llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll:16]: `clang --target=x86_64-linux-gnu` above and the the target triple in IR might be slightly useful to omit `-mtriple` for llc.

Comment (playstation-edd) [reply llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll:16]: I think I've done this - if I got your meaning(?)

File: llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll (hunk 2 above)
Comment (playstation-edd) [reply llvm/test/DebugInfo/X86/debug-aranges-sce-tuning.ll]: Done!

**Pull request merged**